core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -4651,7 +4651,12 @@ public RexNode register(
               currentProjection);
         }
 
-        setRoot(newLeftInput, false);
+        // keep leaves property.
```

Comment:
I don't really understand the comment, can you please elaborate more on that? I think a good explanation of what you are trying to do here is very much needed (i.e., why are we replacing the root in this specific case?).

Reply:
Thanks for your suggestions, the setRoot will replace old root rel by new root, but with a fixed false leaf property. If the old root is a leaf rel, the new root should be a leaf rel, otherwise, the LookupContext will search deeper rel tree leaf node, that will cause wrong field offset.